### ISSUE TITLE
[Blameable][Timestampable] Inconsistent behaviour of 'on="change"' and 'on="update"'

### ISSUE THREAD

**Issue by senaria** [doctrine-extensions/DoctrineExtensions#2023]
There seems to be inconsistent behaviour between `on="change"` and `on="update"`. When persisting a new entity with `change` the field is left as `null`. With 'update` it is filled. Since they are interchangeable i would expect them to function the same.

Let's assume we have the following setup:
```
class A {
	private $name;

    /**
     * @Gedmo\Blameable(on="update")
     */
	private $updatedAt;

    [... some other properties]
}

class B {
	private $name;

    /**
     * @Gedmo\Blameable(on="change", fields={"name"})
     */
	private $updatedAt;

    [... some other properties]
}
```

I want to use `on="update"` for class `A` because I do not want to define a list of every property of class `A`. On the other hand I need to use `on="change"` for class `B` to be able to exclude some properties.

This results in a new record of `A` setting the `updatedAt` but a new record of `B` won't. 

I discovered #1920 which will make `change` also set the value. This would be ok as it is then at least consistent. It would however be preferable to change `update` so that it does not set the variable because technically it is not updated but created. Then it would be useful to allow 2 annotations (`create` and `update`) if you want to also set it on creation.

Can we make them consistent? I would be happy to contribute if you like.

**Comment by AkenRoberts** [comment]
I agree that there is an opportunity for a better solution, since this is a pretty common scenario.

TL;DR: I don't think we can make this work without breaking changes and a new major release, which is unlikely given the current efforts to maintain this library. Try giving `$updatedAt` a value in your entity's constructor when using `change`.

---

Unfortunately the only way I see it happening is by making breaking changes, either to the functionality of the `change` config, or to annotations / configuration to allow multiple events to be configured (which I think is the better solution). For example:

```php
class B {
    /**
     * @var string
     */
    private $name;

    /**
     * @Gedmo\Blameable(onCreate=true)
     *
     * @var \DateTimeInterface
     */
    private $createdAt;

    /**
     * @Gedmo\Blameable(onCreate=true, onChange={"name"})
     *
     * @var \DateTimeInterface
     */
    private $modifiedAt;
}
```

The problem with breaking changes comes down to people resources and maintenance. The original authors of Doctrine Extensions have all moved on, and the couple of us here trying to keep up with issues and occasional bugs can only do so much with the existing library. A new major release just doesn't seem likely.

Giving `$updatedAt` a default value in your entity's constructor _should_ give it a value when using `on="change"`, while retaining the other on-change event functionality. If that doesn't work, let me know.

Otherwise, a couple of us are trying to make some headway on a brand new modern version over at https://github.com/doctrine-extensions/doctrine-extensions, if you're interested in contributing your thoughts or code!

**Comment by yakobe** [comment]
I agree that breaking changes are needed. That's not a bad thing... things move forward and old things get retired. 👍 

I find it a shame that a new repo is needed to achieve it. With semver this should be possible here. Even if it starts to rapidly increase major version numbers; people can always use an older version until they are able to upgrade. In the long run people will be happier.

**Comment by AkenRoberts** [comment]
It could definitely be done here still, there just isn't anyone willing to put the time in.

**Comment by oojacoboo** [comment]
I've just pushed a commit on this @AkenRoberts.  I'm not sure how to turn it into a pull request.  I tried and it resulted in a pull request with multiple commits from a fork.

https://github.com/Atlantic18/DoctrineExtensions/commit/abbe63aa60e6a2f8ba47ba48b326c714bad4f170

I've added support for a new `remove` listener for the annotation on the Blameable extension.  I think this should solve all of the issues people have been having here.  I know it does for us.

**Comment by AkenRoberts** [comment]
@franmomu This might be worth revisiting whenever there's focus on significant updates 👍🏼
